app/views/topics/index.html.erb

```diff
@@ -2,22 +2,5 @@
 
 <div class="row">
   <div class="col-md-8">
-    <% @topics.each do |topic| %>
-      <div class="media">
-        <div class="media-body">
-          <h4 class="media-heading">
-            <%= link_to topic.name, topic %>
-          </h4>
-          <small>
-            <%= topic.description %>
-          </small>
-        </div>
-      </div>
-      <% end %>
-    </div>
-  <% if user_is_authorized_for_topics? %>
-    <div class="col-md-4">
-      <%= link_to "New Topic", new_topic_path, class: 'btn btn-success' %>
-    </div>
-  <% end %>
+    <%= render partial: 'topic', locals: { topic: @topic } %>
```

Comment:
Note this local isn't used in your partial.  Also there is no `@topic` in your controller for this action.  Only `@topics`.